03week/towersOfHanoi.js

```diff
@@ -7,38 +7,84 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+//whiteboard for game at bottom of tests
+
+//main board and object of the game
 let stacks = {
   a: [4, 3, 2, 1],
   b: [],
   c: []
 };
 
+//prints game board in console
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+//moves numbers from stack to stack - a, b, or c
+function movePiece(startStack, endStack) {
+  //move numbers to other stacks
+  //Object.keys(?).pop(?).push(?).shift(?)
+  if (startStack === 'a' && endStack === 'b') {
+    stacks.b.push(stacks.a.pop());
+  } else if (startStack === 'a' && endStack === 'c') {
+    stacks.c.push(stacks.a.pop());
+  } else if (startStack === 'b' && endStack === 'a') {
+    stacks.a.push(stacks.b.pop());
+  } else if (startStack === 'b' && endStack === 'c') {
+    stacks.c.push(stacks.b.pop());
+  } else if (startStack === 'c' && endStack === 'a') {
+    stacks.a.push(stacks.c.pop());
+  } else if (startStack === 'c' && endStack === 'b') {
+    stacks.b.push(stacks.c.pop());
+  } else {
+    console.log('Invalid Entry - Please Input a, b, or c')
+  }
 }
 
-function isLegal() {
-  // Your code here
-
+//checks to make sure game move is legal within the rules of the game
+function isLegal(startStack, endStack) {
+  //is this a legal move
+  //true/false?;
+  if ((stacks[endStack].length === 0) || (stacks[endStack].slice(-1) > stacks[startStack].slice(-1))) {
+    return true;
+  } else {
+    return false;
+  }
 }
 
+//checks to see if all numbers have been moved to stack b or c in order
 function checkForWin() {
-  // Your code here
-
+  //checking for win
+  if ((stacks.b.length === 4) || (stacks.c.length === 4)) {
+    console.log('You Win!!!!!!!!');
+    stacks = {
```

Comment:
towersOfHanoi should take care of this, functional programing, checkForWin should ONLY be responsible for a checking for the win